src/Raven.Studio/typescript/viewmodels/viewModelBase.ts

```diff
@@ -66,7 +66,21 @@ class viewModelBase {
         setTimeout(() => viewModelBase.showSplash(this.isAttached === false), 700);
         this.downloader.reset();
 
-        return this.databasesManager.activateBasedOnCurrentUrl();
+        const dbNameFromUrl = appUrl.getDatabaseNameFromUrl();
+        if (dbNameFromUrl) {
+            const dbAccessLevel = accessManager.default.getEffectiveDatabaseAccessLevel(dbNameFromUrl);
+            const requiredAccessForView = router.activeInstruction().config.requiredAccess;
+            const isViewAllowed = this.allowToAccessView(dbAccessLevel, requiredAccessForView);
+            
+            return this.databasesManager.activateBasedOnCurrentUrl(dbNameFromUrl, isViewAllowed);
+        } else {
+            return true;
+        }
+    }
+
+    protected allowToAccessView(dbAccessLevel: databaseAccessLevel, requiredAccessForView: accessLevel) {
```

Comment:
I think we shoulnd't have database access level here, but generic access level
+ database name as param?

Reply:
I don't think we should have required access a parameter of this method...

method should go more or less like this:
protected canAccessView(accessLevel: accessLevel): boolean {
   // in base class fetch required access level for view and check if access is possible
   // but in base classes we can have different logic - all we have in question is: can user with provided (via accessLevel params) access given view? 
  // this method should get all deps like required level - we shouldn't pass that from outside. 
}

Reply:
We now have:  `protected canAccessView(dbName: string) `